frappe/public/js/frappe/ui/field_group.js

```diff
@@ -86,8 +86,13 @@ frappe.ui.FieldGroup = frappe.ui.form.Layout.extend({
 			var f = this.fields_dict[key];
 			if(f.get_value) {
 				var v = f.get_value();
-				if(f.df.reqd && is_null(v))
+				if(f.df.reqd && is_null(v)) {
 					errors.push(__(f.df.label));
+				}
+
+				if ((f.df.reqd && f.df.fieldtype == "Check") && v == 0) {
```

Comment:
Checkbox only have two states and 0 is often the default value. So this will trigger all the time unless `ignore_errors` is set to true. Can you share use cases where this is useful?

Reply:
@ankush If a checkbox is defined as required / mandatory, it has to be checked (value 1) for the user to continue. This is needed in any case, however, in my specific case, I needed it for a web form, that has checkboxes, that are required to submit the form.

Reply:
Ah, I see. The same function is also used on Desk side, where "mandatory" does not mean it should be 1. On Desk side "mandatory" just means the field should have some input (which by default is there for checkboxes, usually 0)

TBH this feels like a hack to implement "consent" or checkboxes in web facing forms.

Currently, you can add this type of validation to web forms using client side scripts: https://frappeframework.com/docs/user/en/guides/portal-development/web-forms#custom-validation 

But I agree it would be better if there was a simple way to do it.